experimental/dds/tree2/src/feature-libraries/modular-schema/view.ts

```diff
@@ -99,13 +98,7 @@ export class ViewSchema {
 		// and its impossible for an adapter to be correctly implemented if its output type is never
 		// (unless its input is also never).
 		for (const adapter of this.adapters?.tree ?? []) {
-			if (
-				isNeverTree(
-					this.policy,
-					this.schema,
-					this.schema.treeSchema.get(adapter.output) ?? neverTree,
-				)
-			) {
+			if (isNeverTree(this.policy, this.schema, this.schema.treeSchema.get(adapter.output))) {
```

Comment:
Should this ever be a neverTree? I see you've added that API to some tests but the PR description sounds like we def don't want it in production code anymore.

Reply:
undefined (the case where there is not an entry in this map) is not considered a never tree in this code. This is the same as how it was done before, except we have just added one more way to represent never trees in this code (as undefined).

During schema changes, its quite reasonable to end up with neverTrees transiently even without the undefined case. For example, you can add a node schema before you add the types for its children. If the children are in value fields this results in an intermediate state that has a neverTree, which is fine. Or if you are removing types from a schema, you can end up with a type thats never due to being unconditionally recursive before you add back in some alternative types for its children.

Mainly we don't want developers implicitly getting never schema when they have a bug causing their schema to be missing, but since the coder here is all about detecting missing schema and handeling them, and this also works with stores schema not just view schema, it still makes sense for never schema to sometimes occure where without being a crash.